### PR TITLE
Refactor and add sweref99 output

### PR DESCRIPTION
**Pull request by adybbroe**
This PR adds new functionality to output the detections in another custom defined geoid or map projection, other than the standard WGS84. This option is only implemented for the "national" detections, and not for the "regional" ones.

The output geoJSON filenames will have the geoid/projection name specified in the configuration also as part of the filename. So, the standard (non-reprojected WGS84 filename looks like this:
`AFIMG_j01_d20210414_t112643.geojson`

The remproject filename will then be: `AFIMG_j01_d20210414_t112643_<projname>.geojson` where `<projname>` is the name given  in the configuration.

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Tests passed: Passes ``pytest`` 
 - [x] Passes ``flake8`` 
 - [ ] Fully documented

### CODE REVIEW

File: activefires_pp/post_processing.py [hunk 1]
```diff
@@ -431,6 +391,31 @@ def check_incoming_message_and_get_filename(self, msg):
 
         return filename
 
+    def _national_save_and_publish(self, feature_collection, ndata, af_shapeff, msg, sweref99=False):
+        """Take the fearure collection and store the results in a Geojson file and publish."""
```
Comment (mraspaud) [inline activefires_pp/post_processing.py:395]:
```suggestion
        """Take the feature collection and store the results in a Geojson file and publish."""
```

File: activefires_pp/post_processing.py [hunk 2]
```diff
@@ -357,6 +316,7 @@ def __init__(self, configfile, shp_borders, shp_mask, regional_filtermask=None):
         self.output_topic = self.options['publish_topic']
         self.infile_pattern = self.options.get('af_pattern_ibands')
         self.outfile_pattern_national = self.options.get('geojson_file_pattern_national')
+        self.outfile_pattern_national_sweref99 = self.options.get('geojson_file_pattern_national_sweref99')
```
Comment (mraspaud) [inline activefires_pp/post_processing.py:319]: Is there a way to make this less specific? From what I understand the idea is to switch ellipsoids, so what about passing the desired output ellipsoid as an option instead?
Regarding the output file, we could imagine having in the config a list of output patterns with optional ellipsoid for example?

**Review by mraspaud — CHANGES_REQUESTED** [review]
Some genericity issues, but looks good otherwise.

File: activefires_pp/post_processing.py (hunk 2 above)
Comment (adybbroe) [reply activefires_pp/post_processing.py:319]: Ok, yes I see, I think I already contemplated something like this, but went for the simple and fast (though less flexible) solution. Let's discuss...

Comment (adybbroe) [reply activefires_pp/post_processing.py]: Ok, done. Ok?

**Comment by adybbroe** [comment]
@mraspaud What do you think, it is okay now?

File: activefires_pp/post_processing.py [hunk 3]
```diff
@@ -391,25 +393,25 @@ def check_incoming_message_and_get_filename(self, msg):
...
             return
 
         fmda = af_shapeff.metadata
-        if sweref99:
-            pout = Parser(self.outfile_pattern_national_sweref99)
+        if projname != 'default':
+            pout = Parser(self.outfile_patterns_national[projname]['geojson_file_pattern'])
         else:
-            pout = Parser(self.outfile_pattern_national)
+            pout = Parser(self.outfile_patterns_national['deafult']['geojson_file_pattern'])
```
Comment (mraspaud) [inline activefires_pp/post_processing.py:408]:
```suggestion
            pout = Parser(self.outfile_patterns_national['default']['geojson_file_pattern'])
```

**Review by mraspaud — APPROVED** [review]
LGTM, nice will all the new tests!

**Pull request merged**